staging/src/k8s.io/apiserver/plugin/pkg/authorizer/webhook/webhook.go

```diff
@@ -66,11 +75,74 @@ type WebhookAuthorizer struct {
 	retryBackoff        wait.Backoff
 	decisionOnError     authorizer.Decision
 	metrics             AuthorizerMetrics
+	matchConditions     []apiserver.WebhookMatchCondition
+	celMapper           *celMapper
+}
+
+type celMapper struct {
+	compilationResults []authorizationcel.CompilationResult
+}
+
+func compile(expressionAccessors []authorizationcel.ExpressionAccessor, env *environment.EnvSet) (*celMapper, error) {
+	compiler := authorizationcel.NewCompiler(env)
+	compilationResults := make([]authorizationcel.CompilationResult, len(expressionAccessors))
+	for i, expressionAccessor := range expressionAccessors {
+		if expressionAccessor == nil {
+			return nil, &cel.Error{
+				Type:   cel.ErrorTypeInvalid,
+				Detail: "compilation error: expressionAccessor is nil",
+			}
+		}
+		expression := expressionAccessor.GetExpression()
+		compilationResult, _ := compiler.CompileCELExpression(expressionAccessor)
+		if compilationResult.Error != nil {
+			return nil, &cel.Error{
+				Type:   cel.ErrorTypeInvalid,
+				Detail: fmt.Sprintf("compilation error: expression '%s' resulted in error: %v", expression, compilationResult.Error),
+			}
+		}
+		if compilationResult.Program == nil {
+			return nil, &cel.Error{
+				Type:   cel.ErrorTypeInternal,
+				Detail: fmt.Sprintf("compilation error: unexpected internal error compiling expression '%s'", expression),
+			}
+		}
+		compilationResults[i] = compilationResult
+	}
+	return &celMapper{
+		compilationResults: compilationResults,
+	}, nil
+}
+
+func (c *celMapper) eval(ctx context.Context, r *authorizationv1.SubjectAccessReview) ([]authorizationcel.EvaluationResult, error) {
```

Comment:
```suggestion
func (c *celMapper) eval(ctx context.Context, r *unstructured.Unstructured) ([]authorizationcel.EvaluationResult, error) {
```

See if we can convert the request once outside and pass it as input to `eval`.